rusk/src/lib/http/event.rs

```diff
@@ -0,0 +1,228 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use serde::{Deserialize, Serialize};
+use serde_with;
+use serde_with::serde_as;
+use std::fmt::{Display, Formatter};
+
+/// A request sent by the websocket client.
+#[derive(Debug, Serialize, Deserialize)]
+pub(crate) struct WsRequest {
+    pub(crate) headers: serde_json::Map<String, serde_json::Value>,
+    pub(crate) target: WsTarget,
+    // pub(crate) target_type: u8,
+    // pub(crate) target: String,
+    pub(crate) topic: String,
+    pub(crate) data: DataType,
+}
```

Comment:
**Suggestion:** by making the struct itself `pub(crate)` we can make the fields just `pub` for brevity.
```suggestion
/// A request sent by the websocket client.
#[derive(Debug, Serialize, Deserialize)]
pub(crate) struct WsRequest {
    pub headers: serde_json::Map<String, serde_json::Value>,
    pub target: WsTarget,
    // pub target_type: u8,
    // pub target: String,
    pub topic: String,
    pub data: DataType,
}
```